packages/@aws-cdk/aws-iam/test/integ.imported-role.ts

```diff
@@ -0,0 +1,17 @@
+import { App, Stack } from '@aws-cdk/core';
+import {
+  IAM_IMPORTED_ROLE_STACK_SAFE_DEFAULT_POLICY_NAME,
+} from '@aws-cdk/cx-api';
+import { PolicyStatement, Role, ServicePrincipal } from '../lib';
+
+const app = new App({ context: { [IAM_IMPORTED_ROLE_STACK_SAFE_DEFAULT_POLICY_NAME]: true } });
+
+const stack = new Stack(app, 'integ-iam-imported-role-1');
+
+const role = new Role(stack, 'TestRole', {
+  assumedBy: new ServicePrincipal('sqs.amazonaws.com'),
+});
+
+role.addToPolicy(new PolicyStatement({ resources: ['*'], actions: ['sqs:SendMessage'] }));
+
+app.synth();
```

Comment:
Please make sure you follow the instructions in [INTEGRATION_TESTS.md](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)

You're missing this bit: `new integ.IntegTest(...)`

Reply:
Thanks! I copied from another test and missed that.

Reply:
I've added the `IntegTest`, but wanted to reconfirm with you, as this would then be the only integ test in `aws-iam` using `IntegTest`.

Reply:
To be honest, I'm also not sure how I would use `IntegTest` when I have multiple stacks that all are needed for the same test (as I need in the rewritten version of the test), as `IntegTest` takes test stacks and, while the code doesn't look like it yet, they might be meant to run independently. I'll add a separate commit for this so that it is easy to revert if needed.

Reply:
This is a nice example:

https://github.com/aws/aws-cdk/blob/9f41881be1cca31129078b7839dded480220f060/packages/@aws-cdk/aws-iam/test/integ.permissions-boundary.ts

The stack with the role is the test subject. The extra stack are support. Finally we need to pass make sure the assertions stack depends on both support stacks.

Reply:
I'm not sure this works as there now are circular dependencies (the supporting stacks need to get the roleName from the role stack, but the role stack explicitly depends on the other two). Instead, I have the role2 stack as the stack-under-test, it shouldn't make a difference for the integration test which stack is under test, as long as all of them are deployed.